examples_gbs/run_tutorial_similarity.py

```diff
@@ -0,0 +1,203 @@
+# number of 5:
+
+print(list(similarity.orbits(5)))
+
+##############################################################################
+# All 5-photon samples fall into one of the orbits above. A 5-photon event with
+# ``max_count_per_mode = 3`` means that we combine the orbits: ``[[1, 1, 1, 1, 1], [2, 1, 1, 1],
+# [3, 1, 1], [2, 2, 1], [3, 2]]`` and ignore the orbits ``[[4, 1], [5]]``. For example,
+# the sample ``[0, 0, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 3, 0, 0, 0]`` is a 5-photon event:
+
+print(similarity.sample_to_event([0, 0, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 3, 0, 0, 0], 3))
+
+##############################################################################
+# Samples with more than ``max_count_per_mode`` in any mode are not counted as part of the event:
+
+print(similarity.sample_to_event([0, 4, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0], 3))
+
+##############################################################################
+# Now that we have mastered orbits and events, how can we make a feature vector? It was shown in
+# :cite:`schuld2019quantum` that one way of making a feature vector of a graph is to measure the
+# probabilities of events. Specifically, for a :math:`k` photon event :math:`E_{k, n_{\max}}`
+# with maximum count per mode :math:`n_{\max}` and corresponding probability :math:`p_{k,
+# n_{\max}}:=p_{E_{k, n_{\max}}}(G)` with respect to a graph :math:`G`, a feature vector can be
+# written as
+#
+# .. math::
+#     f_{\mathbf{k}, n_{\max}} = (p_{k_{1}, n_{\max}}, p_{k_{2}, n_{\max}}, \ldots , p_{k_{K}, n_{\max}}),
+#
+# where :math:`\mathbf{k} := (k_{1}, k_{2}, \ldots , k_{K})` is a list of different total photon
+# numbers.
+#
+# Calculating a feature vector
+# ----------------------------
+#
+# We provide two methods for calculating a feature vector of GBS event probabilities in
+# Strawberry Fields:
+#
+# 1. Through sampling
+# 2. Using a Monte Carlo estimate of the probability
+#
+# For the first method, all one needs to do is generate some GBS samples from the graph of
+# interest and feed through:
+
+print(similarity.feature_vector_sampling(m0, event_photon_numbers=[2, 4, 6], max_count_per_mode=2))
+
+##############################################################################
+# For the second method, suppose we want to calculate the event probabilities exactly rather than
+# through sampling. To do this, we consider the event probability :math:`p_{k, n_{\max}}` as the
+# sum over all sample probabilities in the event. As discussed in the GBS `tutorial
+# <gaussian_boson_tutorial>`, each sample probability is determined by the hafnian of a relevant
+# sub-adjacency matrix. While this is tough to calculate, what makes calculating :math:`p_{k,
+# n_{\max}}` really tough is the number of samples the corresponding event contains! For example,
+# the 17-photon event :math:`E_{k=6, n_{\max}=2}` contains the following number of samples:

```

Comment:
The term "17-photon event" might be confused with ":math:`k` photon event" appeared above. Is there a better way to refer to the 17? (maybe modes?)